packages/@o3r/rules-engine/src/stores/rulesets/rulesets.selectors.ts

```diff
@@ -30,6 +30,34 @@ export const selectRulesetsStorePendingStatus = createSelector(selectRulesetsSta
  */
 const isValidDate = (d: any) => !isNaN(d) && d instanceof Date;
 
+/**
+ * Checks if the current date and time fall within the validity range specified in the ruleSet.
+ *
+ * @param ruleSet The set of rules containing the validity range.
+ */
+function checkValidityRange(ruleSet: Ruleset) {
+  const validity = ruleSet.validityRange;
+  if (!validity || !validity.from && !validity.to) {
+    return true;
+  }
+
+  const from = validity.from && new Date(validity.from);
+  const to = validity.to && new Date(validity.to);
+
+  if (to && !isValidDate(to) || from && !isValidDate(from)) {
+    return false;
+  }
+  const time = Date.now();
+  if (to && from) {
+    return from.getTime() <= time && to.getTime() >= time;
+  }
+
+  if (from) {
+    return from.getTime() <= time;
+  }
+  return to && to.getTime() >= time;
+}
+
 /**
  * Returns only the rulesets ids which are not onDemand and in the validity range
  * OnDemand takes precedence over validity range.
```

Comment:
It seems that the current behavior was planned 
if after clarification we want to change it we should update the documentation accordingly

Reply:
JS doc updated. The `docs/` part has to be done in another PR on main because it does not exist on 10.4 yet